apps/ocpp/admin/charger/diagnostics.py

```
@@ -0,0 +1,355 @@
+            raise self.DiagnosticsDownloadError(_("Unable to resolve diagnostics host: %s") % exc) from exc
+
+        for _, _, _, _, sockaddr in addrinfos:
+            candidate = ipaddress.ip_address(sockaddr[0])
+            if candidate.is_private or candidate.is_loopback or candidate.is_link_local or candidate.is_reserved or candidate.is_multicast:
+                raise self.DiagnosticsDownloadError(_("Diagnostics location resolves to a disallowed private address."))
+        return hostname
+
+    def _download_diagnostics(
+        self,
+        request,
+        charger: Charger,
```

Ruff (0.15.0)

[error] `Charger` may be undefined, or defined from star imports

(F405)